common/src/app/common/geom/shapes/intersect.cljc

```diff
@@ -335,13 +337,32 @@
   (let [lines (grc/rect->lines rect)]
     (is-point-inside-evenodd? point lines)))
 
-(defn has-point?
-  "Check if the shape contains a point"
+(defn slow-has-point?
   [shape point]
-  (let [lines (points->lines (:points shape))]
-    ;; TODO: Will only work for simple shapes
+  (let [lines (points->lines (dm/get-prop shape :points))]
     (is-point-inside-evenodd? point lines)))
 
+(defn fast-has-point?
```

Comment:
Will this give the correct result? This is checking if the point is inside the bounding rect of the shape. But for circles, for example, a point may be inside the rect but outside the shape (e.g. in a corner).

Reply:
Agree, and I will fix it. This change is because this function is mostly used for frames, which behave like rects.
Also, this function does not covers rotation, but this is how it worked until now so I don't go to change it.

Reply:
Added circle to the slow path